modules/gapi/test/streaming/gapi_streaming_tests.cpp

```diff
@@ -302,6 +302,104 @@ TEST_P(GAPI_Streaming, SmokeTest_VideoConstSource_NoHang)
     EXPECT_EQ(ref_frames, test_frames);
 }
 
+TEST_P(GAPI_Streaming, SmokeTest_AutoMeta)
+{
+    cv::GMat in;
+    cv::GMat in2;
+    cv::GMat roi = cv::gapi::crop(in2, cv::Rect{1,1,256,256});
+    cv::GMat blr = cv::gapi::blur(roi, cv::Size(3,3));
+    cv::GMat out = blr - in;
+
+    auto testc = cv::GComputation(cv::GIn(in, in2), cv::GOut(out))
+        .compileStreaming(cv::compile_args(cv::gapi::use_only{GetParam()}));
+
+    cv::Mat in_const = cv::Mat::eye(cv::Size(256,256), CV_8UC3);
+    cv::Mat tmp;
+
+    // Test with one video source
+    auto in_src = gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(findDataFile("cv/video/768x576.avi"));
+    testc.setSource(cv::gin(in_const, in_src));
+    testc.start();
+
+    std::size_t test_frames = 0u;
+    while (testc.pull(cv::gout(tmp))) test_frames++;
+    EXPECT_EQ(100u, test_frames);
+
+    // Now test with another one
+    in_src = gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(findDataFile("cv/video/1920x1080.avi"));
+    testc.setSource(cv::gin(in_const, in_src));
+    testc.start();
+
+    test_frames = 0u;
+    while (testc.pull(cv::gout(tmp))) test_frames++;
+    EXPECT_EQ(165u, test_frames);
+}
+
+
+TEST_P(GAPI_Streaming, SmokeTest_Two_Const_Mats)
+{
+    cv::GMat in;
+    cv::GMat in2;
+    cv::GMat roi = cv::gapi::crop(in2, cv::Rect{1,1,256,256});
+    cv::GMat blr = cv::gapi::blur(roi, cv::Size(3,3));
+    cv::GMat out = blr - in;
+
+    auto testc = cv::GComputation(cv::GIn(in, in2), cv::GOut(out))
+        .compileStreaming(cv::compile_args(cv::gapi::use_only{GetParam()}));
+
+    cv::Mat in_const = cv::Mat::eye(cv::Size(256,256), CV_8UC3);
+    cv::Mat tmp;
+
+    // Test with firs image
+    auto in_src = cv::imread(findDataFile("cv/edgefilter/statue.png"));
+    testc.setSource(cv::gin(in_const, in_src));
+    testc.start();
+
+    ASSERT_TRUE(testc.pull(cv::gout(tmp)));
+
+    testc.stop();
+
+    // Now test with second image
+    in_src = cv::imread(findDataFile("cv/edgefilter/kodim23.png"));
+    testc.setSource(cv::gin(in_const, in_src));
+    testc.start();
+
+    ASSERT_TRUE(testc.pull(cv::gout(tmp)));
+
+    testc.stop();
+}
+
+TEST_P(GAPI_Streaming, SmokeTest_One_Video_One_Const_Scalar)
+{
+    cv::GMat in_m;
+    cv::GScalar in_s;
+    cv::GMat out_m = in_m * in_s;
+
+    auto testc = cv::GComputation(cv::GIn(in_m, in_s), cv::GOut(out_m))
+        .compileStreaming(cv::compile_args(cv::gapi::use_only{GetParam()}));
+
+    cv::Mat tmp;
+    // Test with one video source and scalar
+    auto in_src = gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(findDataFile("cv/video/768x576.avi"));
+    cv::Scalar sc = 32;
```

Comment:
32 is a not good value to multiply on. For any pixel with intensity > 8 your multiplication will saturate to 255 quickly. Use a pair of 1.25 and 0.75 instead, for example.

Reply:
fixed